### ISSUE TITLE
Accidental space issues are not detected by the `valid-title` rule with type checking enabled

### ISSUE THREAD

**Issue by TimMoore** [vitest-dev/eslint-plugin-vitest#748]
With a configuration like this in `eslint.config.mjs`:

```javascript
  {
    files: ['**/*.test.ts', '**/*.test.tsx', '**/__tests__/**/*', '**/*.spec.ts', '**/*.spec.tsx'],
    plugins: { vitest },
    rules: {
      ...vitest.configs.recommended.rules,
    },
    settings: {
      vitest: {
        typecheck: true,
      },
    },
  },
```

A title like this, should trigger an  `accidentalSpace` error, but does not:

```typescript
describe(' extra spaces ', () => { /* ... */ });
```

It _does_ fail if `typecheck: true` is commented out in the configuration:

```text
  30:12  error  should not have leading or trailing spaces  vitest/valid-title
```

This can also be reproduced by adding tests like this to the `valid-title.test.ts` invalid examples:

```typescript
    {
      code: 'describe(" foo", function () {})',
      settings: { vitest: { typecheck: true } },
      output: 'describe("foo", function () {})',
      errors: [{ messageId: 'accidentalSpace', column: 10, line: 1 }],
    },
```

**Comment by TimMoore** [comment]
I've pushed a failing test to my fork at https://github.com/TimMoore/eslint-plugin-vitest/commit/be85a25e96b87a3088c47306fe2b5b095f6bca58 that demonstrates the problem.

I had a quick crack at a fix, but it broke other tests. I'll try to give it a closer look when I have some free time, but if anyone else wants to fix it before I get a chance, great!
